app/models/order_detail.rb

```diff
@@ -845,16 +846,18 @@ def manual_fulfilled_at=(string)
   end
 
   def time_data
-    @time_data ||= product.time_data_for(self)
+    if product.respond_to?(:time_data_for)
```

Comment:
Not sure I love it yet: I might go back to more of the constant lookup. Still need to explore, but this works for now.